### DISCUSSION TITLE
Should we rename the block operation `subvector()` to `slice()`?

### DISCUSSION THREAD

**Discussion by weslleyspereira** [tlapack/tlapack#26] [Q&A] [answered]
Should we rename the block operation `subvector()` to `slice()`? A suggestion from @thijssteel.
And maybe `submatrix()` could be called `block()` as in Eigen.

The idea is to reduce the wordiness of the code.

**Comment by thijssteel** [comment, marked as answer]
My suggestion was to use `slice()` for submatrix, subvector and column/row selection.

You explained that you don't want to do that because it would mess with the vector access [i] vs matrix access A(i,j), but I don't really get why that would be.

I was thinking something along the lines of:

```c++
vector_t slice( vector_t v, pair_t p );

vector_t slice( matrix_t A, pair_t p, index_t j );

vector_t slice( matrix_t A, index_t i, pair_t p );

matrix_t slice( matrix_t A, pair_t pi, pair_t pj );
```
I don't see how this forces us to change the vector or matrix access. It's basically the same as using `slice()` for vectors and `block` for matrices, you just overload the function.

But I don't feel that strongly about this, `block` is fine too.

**Reply by weslleyspereira** [reply]
Hi. Thanks for explaining it better!

I agree with:

> ```c++
> vector_t slice( vector_t v, pair_t p );
> 
> vector_t slice( matrix_t A, pair_t p, index_t j );
> 
> vector_t slice( matrix_t A, index_t i, pair_t p );
> ```

These last two functions would avoid things like `subvector( col( A, i ), pair{i+1,m} )`. We would be able to use `slice( A, pair{i+1,m}, i )`. Then, if we have those 3, I also agree it would make sense to have:

> ```c++
> matrix_t slice( matrix_t A, pair_t pi, pair_t pj );
> ```
